dsl/pace/dsl/dace/orchestration.py

```diff
@@ -146,11 +153,38 @@ def _build_sdfg(
         with DaCeProgress(config, "Simplify (2/2)"):
             sdfg.simplify(validate=False, verbose=True)
 
+        # Remove all Persistent memory
+        with DaCeProgress(config, "Turn Persistents into pooled Scope"):
+            memory_pooled = 0.0
+            for _sd, _aname, arr in sdfg.arrays_recursive():
+                if arr.lifetime == dace.AllocationLifetime.Persistent:
+                    arr.pool = True
+                    memory_pooled += arr.total_size * arr.dtype.bytes
+                    arr.lifetime = dace.AllocationLifetime.Scope
+            memory_pooled = float(memory_pooled) / (1024 * 1024)
```

Comment:
Is my understanding correct that this is the moment when the arrays are switched to using pooled memory?

Reply:
Yes. DaCe will automatically, at code generation, pool all Scoped arrays flagged. So we swap persistent arrays (e.g. arrays in sub-SDFG not passed as parameters to the top SDFG) into scope, and flag them.
Actual pooling is done at code gen time

Reply:
I am clarifying in the comment